crates/core/tedge_actors/src/actors.rs

```diff
@@ -37,6 +37,91 @@ impl<S: Server> ServerActor<S> {
     }
 }
 
+/// A server defines the behavior of an actor processing requests sending back responses.
+///
+/// A `Server` is defined by:
+/// - a `Request` message type for the requests received from client actors,
+/// - a `Response` message type for the responses sent back to the requesters,
+/// - an asynchronous `handle` method that defines how the server responds to a request,
+///   updating its state and possibly performing side effects.
+///
+/// ```
+/// # use crate::tedge_actors::{Server, MessageBox, SimpleMessageBox};
+/// # use async_trait::async_trait;
+///
+/// # use crate::tedge_actors::examples;
+/// # type Operation = examples::calculator::Operation;
+/// # type Update = examples::calculator::Update;
+///
+/// /// State of the calculator server
+/// #[derive(Default)]
+/// struct Calculator {
+///     state: i64,
+/// }
+///
+/// /// Implementation of the calculator behavior
+/// #[async_trait]
+/// impl Server for Calculator {
+///
+///     type Request = Operation;
+///     type Response = Update;
+///
+///     fn name(&self) -> &str {
+///         "Calculator"
+///     }
+///
+///     async fn handle(&mut self, request: Self::Request) -> Self::Response {
+///         // Act accordingly to the request
+///         let from = self.state;
+///         let to = match request {
+///            Operation::Add(x) => from + x,
+///            Operation::Multiply(x) => from * x,
+///         };
+///
+///         // Update the server state
+///         self.state = to;
+///
+///         // Return the response
+///         Update{from,to}
+///     }
+/// }
+/// ```
+///
+/// To be used as an actor, a `Server` is wrapped into a [ServerActor](crate::ServerActor)
+///
+/// ```
+/// # use tedge_actors::{Actor, MessageBox, ServerActor, SimpleMessageBox};
+/// # use crate::tedge_actors::examples::calculator::*;
+/// #
+/// # #[tokio::main]
+/// # async fn main_test() {
+/// #
+/// // Create an actor to handle the requests to a server
+/// let server = Calculator::default();
+/// let actor = ServerActor::new(server);
+///
+/// // As for any actor, one needs a bidirectional channel to the message box of the server.
+/// let (mut test_box, actor_box) = SimpleMessageBox::channel("Test", 10);
```

Comment:
```suggestion
/// let (mut client_box, server_box) = SimpleMessageBox::channel("Test", 10);
```
Just to highlight our server/client notion more explicitly.

Reply:
Done